apps/ejabberd/src/ejabberd_listener.erl

```diff
@@ -190,16 +193,24 @@ listen_tcp(PortIPProto, Module, SockOpts, Port, IPS) ->
             end
     end.
 
+override_sock_opts([], Opts) ->
+    Opts;
+override_sock_opts([Override | OverrideOpts], Opts) ->
+    NewOpts = do_override(Override, Opts),
+    override_sock_opts(OverrideOpts, NewOpts).
+
+do_override({ip, _} = IP, Opts) ->
+    lists:keystore(ip, 1, Opts, IP);
+do_override({backlog, _} = Backlog, Opts) ->
+    lists:keystore(backlog, 1, Opts, Backlog);
+do_override(inet6, Opts) ->
+    [inet6 | lists:delete(inet6, Opts)];
+do_override(inet, Opts) ->
+    [inet | lists:delete(inet, Opts)];
```

Comment:
I like this explicit solution! One minor nitpick:

Isn't
```[inet6 | lists:delete(inet6, Opts)]```
the same thing as
```Opts```?

Also, it's not technically possible to **delete** one of the `inet|inet6` options now. Do we need those clauses? Maybe just erroring out with `error({cannot_override_option, O, Opts})` would be more straightforward for the user?